src/resmom/requests.c

```diff
@@ -619,23 +622,25 @@ req_deletejob(struct batch_request *preq)
 	 * mom_deljob_wait() sets substate to
 	 * prevent sending more OBIT messages
 	 */
+	pjob->ji_preq = preq;
+	hook_input = (mom_hook_input_t *) malloc (sizeof(mom_hook_input_t));
+	if (hook_input == NULL) {
+		log_err(errno, __func__, MALLOC_ERR_MSG);
+		return;
+	}
+	mom_hook_input_init(hook_input);
+	hook_input->pjob = pjob;
 
-#if MOM_ALPS
-	pjob->ji_preq = preq;	/* needed when canceling ALPS */
-#else
-	pjob->ji_preq = NULL;
-#endif
-	mom_hook_input_init(&hook_input);
-	hook_input.pjob = pjob;
-
-	mom_hook_output_init(&hook_output);
-	hook_output.reject_errcode = &reject_errcode;
-	hook_output.last_phook = &last_phook;
-	hook_output.fail_action = &hook_fail_action;
+	if (mom_process_hooks(HOOK_EVENT_EXECJOB_END,
+		PBS_MOM_SERVICE_NAME, mom_host, hook_input,
+		hook_output, hook_msg, sizeof(hook_msg), 1) == HOOK_RUNNING_IN_BACKGROUD)
```

Comment:
this should be HOOK_RUNNING_IN_BACKGROUND.

Reply:
Done